java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/htmlui/Browser.java

```diff
@@ -18,538 +18,167 @@
-                );
-            }
-            w.write(""
-                    + "        (0 || eval)(this.responseText);\n"
-                    + "      } catch (e) {\n"
-                    + "        console.warn(e); \n"
-                    + "      } finally {\n"
-                    + "        waitForCommand(counter + 1);\n"
-                    + "      }\n"
-                    + "    };\n"
-                    + "    request.send();\n"
-                    + "  } catch (e) {\n"
-                    + "    console.warn(e);\n"
-                    + "    waitForCommand(counter + 1);\n"
-                    + "  }\n"
-                    + "}\n"
-                    + "waitForCommand(1);\n"
-            );
-            w.write("  </script>\n");
+    private Map<String, String> getResources(URL[] resources) throws IOException {
+        Map<String, String> ret = new HashMap<>();
+        for (URL resource : resources) {
+            String path = resource.getPath();
+            int idx = path.lastIndexOf("/");
+            ret.put(idx < 0 ? path : path.substring(idx + 1), getText(resource, null));
         }
+        return ret;
     }
 
-    String createCallbackFn(String prefix, String id) {
-        StringBuilder sb = new StringBuilder();
-        sb.append("this.toBrwsrSrvr = function(name, a1, a2, a3, a4) {\n"
-            + "var url = '").append(prefix).append("command.js?id=").append(id).append("&name=' + name;\n"
-            + "var body = 'p0=' + encodeURIComponent(a1);\n"
-            + "body += '&p1=' + encodeURIComponent(a2);\n"
-            + "body += '&p2=' + encodeURIComponent(a3);\n"
-            + "body += '&p3=' + encodeURIComponent(a4);\n"
-            + "var request = new XMLHttpRequest();\n"
-        );
-        if (Browser.this.config.debug) {
-            sb.append(""
-            + "console.log('PUT ... ' + body.substring(0, 80));\n"
-            + "var now = new Date().getTime();\n"
-            );
-        }
-        sb.append(""
-            + "request.open('PUT', url, false);\n"
-            + "request.setRequestHeader('Content-Type', 'application/x-www-form-urlencoded; charset=utf-8');\n"
-            + "request.send(body);\n"
-            + "var txt = request.responseText;\n"
-        );
-        if (Browser.this.config.debug) {
-            sb.append(""
-            + "var then = new Date().getTime();\n"
-            + "if (txt && txt !== 'null') {\n"
-            + "  var cmd = document.getElementById('cmd');\n"
-            + "  if (cmd) cmd.innerHTML = txt.substring(0,80);\n"
-            + "}\n"
-            + "console.log('... PUT [' + (then - now) + 'ms]: ' + txt.substring(0, 80));\n"
-            );
-        }
-        sb.append(""
-            + "return txt;\n"
-            + "};\n"
-        );
-        return sb.toString();
+    private String createCallbackFn(String id) {
+        return "this.toBrwsrSrvr = function(name, a1, a2, a3, a4) {\n"
+             + "  vscode.postMessage({\n"
```

Comment:
You could use `vscode` here and then delete it. Just wrap into 
```
(function() {
  var vscode = globalThis.vscode;
  delete globalThis.vscode;
  
  globalThis.toBrwsrSrvr = ....
)();
```
after deleting `vscode` from global scope the page could even load _potentially untrusted code_.